projects/views.py

```diff
@@ -119,16 +117,18 @@ def accept_invite(request, invite_code):
     user = request.user
     # Check for existing day pass, use this invite instead
     day_pass = get_day_pass(user.id, invitation.project.id)
-    if day_pass is not None:
-        day_pass.delete()
-    if accept_invite_for_user(user, invitation, mapper):
-        messages.success(request, "Accepted invitation")
-        return HttpResponseRedirect(
-            reverse("projects:view_project", args=[invitation.project.id])
-        )
-    else:
-        messages.error(request, invitation.get_cant_accept_reason())
-        return HttpResponseRedirect(reverse("projects:user_projects"))
+    try:
+        if accept_invite_for_user(user, invitation, mapper):
+            messages.success(request, "Accepted invitation")
+            return HttpResponseRedirect(
+                reverse("projects:view_project", args=[invitation.project.id])
+            )
+        else:
+            messages.error(request, invitation.get_cant_accept_reason())
+            return HttpResponseRedirect(reverse("projects:user_projects"))
+    finally:
```

Comment:
This may be the first time I've ever seen a finally block that was actually necessary.